common/src/test/java/io/netty/util/concurrent/FastThreadLocalTest.java

```diff
@@ -75,4 +75,61 @@ public void run() {
             throw t;
         }
     }
+
+    @Test(timeout = 4000)
+    public void testOnRemoveCalledForFastThreadLocal() throws Exception {
+        testOnRemoveCalled(true);
+    }
+
+    @Test(timeout = 4000)
+    public void testOnRemoveCalledForNonFastThreadLocal() throws Exception {
+        testOnRemoveCalled(false);
+    }
+
+    private static void testOnRemoveCalled(boolean fastThreadLocal) throws Exception {
+
+        final TestFastThreadLocal threadLocal = new TestFastThreadLocal();
+        final TestFastThreadLocal threadLocal2 = new TestFastThreadLocal();
+
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                assertEquals(Thread.currentThread().getName(), threadLocal.get());
```

Comment:
A bit late to this PR, but what happens if code sets a FastThreadLocal and the thread exits before it's read? It looks like the cleaner code is only registered on the `get` path.

Reply:
@zpencer good point. Let me fix this.

Reply:
@zpencer https://github.com/netty/netty/pull/7534 PTAL